Client/Frontend/Browser/BrowserViewController/OpenSearch/OpenSearchEngineButton.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import SnapKit
+import BraveShared
+
+// MARK: - OpenSearchEngineButton
+
+class OpenSearchEngineButton: UIView {
+
+    // MARK: State
+    
+    enum State {
+        case loading
+        case enabled
+        case disabled
+    }
+
+    // MARK: Properties
+    
+    var state: State {
+        didSet {
+            switch state {
+            case .disabled:
+                searchButton.isHidden = hidesWhenDisabled ? true : false
+                loadingIndicator.stopAnimating()
+                searchButton.tintColor = UIColor.Photon.grey50
```

Comment:
Will probably need `appearanceTintColor` for iOS 12

Reply:
Fix at https://github.com/brave/brave-ios/pull/3231/commits/ba493be2def981cae70ab0ce8231326645c31a0d